### PR TITLE
Hw2

### PR DESCRIPTION
**Pull request by Le0nRoy**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -56,6 +68,16 @@
             <version>7.1.0</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-java</artifactId>
+            <version>3.141.59</version>
```
Comment (DmitryKhodakovsky) [inline pom.xml:74]: It is better extract versions to the `maven properties`

File: src/test/java/hw2/BaseClass.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @Test
+    public void loginAndCheckUsername() {
+
+        String result = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("user-name"))).getText();
+        String expected = "ROMAN IOVLEV";
+
+        assertEquals(result, expected);
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/BaseClass.java:34]: Why do you need these two tests?

File: src/test/java/hw2/ex1/Exercise1.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import hw2.BaseClass;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.testng.annotations.Test;
+
+import java.util.*;
+
+import static org.testng.Assert.*;
+
+public class Exercise1 extends BaseClass {
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex1/Exercise1.java:12]: Why do you decide split scenario to several tests methods?

File: src/test/java/hw2/ex2/Exercise2.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            }
+        }
+    }
+
+    private void switchToServiceDifferentElements() {
+
+        wait.until(ExpectedConditions.elementToBeClickable(
+                By.xpath("//ul[@class='uui-navigation nav navbar-nav m-l8']//*[contains(text(),'Service')]"))).click();
+        wait.until(ExpectedConditions.elementToBeClickable(
+                By.xpath("//../*[contains(text(),'Different elements')]"))).click();
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex2/Exercise2.java:103]: Locators could be simplified

File: src/test/java/hw2/ex2/Exercise2.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        chromeDriver.findElement(By.xpath("//*[@class='colors']")).click();
+        String expectedOption = "Yellow";
+
+        List<WebElement> elements = chromeDriver.findElements(By.xpath("//*[@class='colors']//option"));
+        for (Iterator<WebElement> iterator = elements.iterator(); iterator.hasNext(); ) {
+            WebElement el = iterator.next();
+            if (el.getText().equals(expectedOption)) {
+                el.click();
+            }
+        }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex2/Exercise2.java:85]: Why do you do not use `Select` element?

File: src/test/java/hw2/ex2/Exercise2.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+        assertEquals(elements.size(), 4);
+        for (Iterator<WebElement> iterator = elements.iterator(); iterator.hasNext(); ) {
+            WebElement el = iterator.next();
+            if ( el.getText().equals(expectedOption) ) {
+                el.click();
+            }
+        }
+
+        elements = chromeDriver.findElements(By.xpath("//*[@class='info-panel-section']//li"));
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex2/Exercise2.java:61]: Locator could be simplified

File: src/test/java/hw2/ex2/Exercise2.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+                assertTrue(el.getText().contains("condition changed to true"));
+            }
+        }
+    }
+
+    @Test
+    private void selectRadiobuttonsAndAssertTheirStatusses() {
+
+        switchToServiceDifferentElements();
+        List<WebElement> elements = chromeDriver.findElements(By.xpath("//*[@class='label-radio']"));
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex2/Exercise2.java:50]: Why do you use xpath?

File: src/test/java/hw2/ex2/Exercise2.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+        assertEquals(elements.size(), 4);
+        for (Iterator<WebElement> iterator = elements.iterator(); iterator.hasNext(); ) {
+            WebElement el = iterator.next();
+            if ( (el.getText().equals(expectedOptions[0])) || (el.getText().equals(expectedOptions[1])) ) {
+                el.click();
+            }
+        }
+
+        elements = chromeDriver.findElements(By.xpath("//*[@class='info-panel-section']//li"));
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex2/Exercise2.java:32]: Why do you use xpath?

File: src/test/java/hw2/ex2/Exercise2.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+import static org.testng.Assert.*;
+
+public class Exercise2 extends BaseClass {
+
+    @Test
+    private void selectCheckboxesAndAssertTheirStatuses() {
+
+        switchToServiceDifferentElements();
+        List<WebElement> elements = chromeDriver.findElements(By.xpath("//*[@class='label-checkbox']"));
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex2/Exercise2.java:21]: Why do you use xpath?

File: src/test/java/hw2/ex2/Exercise2.java [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.Iterator;
+import java.util.List;
+
+import static org.testng.Assert.*;
+
+public class Exercise2 extends BaseClass {
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex2/Exercise2.java:15]: Why do you decide split scenario for the several test methods?

File: src/test/java/hw2/ex1/Exercise1.java [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+        String result = frameButton.getAttribute("value");
+        String expected = "Frame Button";
+
+        assertEquals(result, expected);
+    }
+
+    @Test
+    private void checkItemsOnLeftSection() {
+
+        List<WebElement> elements = chromeDriver.findElements(By.xpath("//ul[@class='sidebar-menu']/li/a/span"));
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex1/Exercise1.java:78]: Locator could be simplified to css

File: src/test/java/hw2/ex1/Exercise1.java [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    @Test
+    private void checkTextsOnIndexPage() {
+
+        List<WebElement> elements = chromeDriver.findElements(By.className("benefit-txt"));
+
+        int expectedNumOfTexts = 4;
+
+        assertEquals(elements.size(), expectedNumOfTexts);
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex1/Exercise1.java:49]: What is the text that you assert in this test?

File: src/test/java/hw2/ex1/Exercise1.java [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+import static org.testng.Assert.*;
+
+public class Exercise1 extends BaseClass {
+
+    @Test
+    private void checkItemsOnHeader() {
+
+        List<WebElement> elements = chromeDriver.findElements(
+                By.xpath("//ul[@class='uui-navigation nav navbar-nav m-l8']/li"));
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex1/Exercise1.java:18]: Could be simplified to css selector

File: src/test/java/hw2/ex1/Exercise1.java (hunk 3 above)
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex1/Exercise1.java:12]: Why do you do not use `SoftAssertion` for the ex1 as it was required for the first tasks?

**Comment by Le0nRoy** [comment]
Resolved all commentaries

File: src/test/java/hw2/ex1/Exercise1.java [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+        expectedNumOfElements = 5;
+        expectedNames = new ArrayList<String>(Arrays.asList(
+                "Home",
+                "Contact form",
+                "Service",
+                "Metals & Colors",
+                "Elements packs"
+        ));
+        softAssert.assertEquals(elements.size(), expectedNumOfElements);
+        softAssert.assertNotSame(elements, expectedNames);
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/ex1/Exercise1.java:83]: You have forgot main method for the soft assert

**Pull request merged**